service/controller/v15/resource/cloudformation/current.go

```diff
@@ -33,33 +32,35 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 	// stack. We dispatch our custom StackState structure and enrich it with all
 	// information necessary to reconcile the cloudformation resource.
 	var stackOutputs []*cloudformation.Output
-	var stackStatus string
 	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "finding the guest cluster main stack outputs in the AWS API")
+
+		var stackStatus string
 		stackOutputs, stackStatus, err = sc.CloudFormation.DescribeOutputsAndStatus(stackName)
 		if cloudformationservice.IsStackNotFound(err) {
-			r.logger.LogCtx(ctx, "level", "debug", "message", "did not find the guest cluster main stack in the AWS API")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "did not find the guest cluster main stack outputs in the AWS API")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "the guest cluster main stack is does not exist")
```

Comment:
There is already a log message stating that: main stack was not found.

Reply:
There is a message about `stack outputs` not being found. The reason for that is that the stack does not exits. There may be different reasons. E.g. stack status is in progress.
Does that make sense?

Reply:
In a perfect world the logs would accurately describe the reason.

Reply:
What do you mean? Is there something to improve?

Reply:
Never mind. I think I just read your comment wrong.

Reply:
Ack. To be clear. Different reasons are accurately (best effort) described in other `else if` branches.

Reply:
My point here was that we have two log lines, which sounds very similar to me:
- "did not find the guest cluster main stack outputs in the AWS API"
- "the guest cluster main stack is does not exist"

They both inform us that the guest cluster stack is not there, I am not a big fan of very verbose logs and I do not see what additional information the second line brings here.
IMO First line is enough to understand what is going.